### PR TITLE
Blog details page

### LINKED ISSUE #1198: Blog detail templates
Requires #1199 for completion, but can be done in parallel

- [ ] Article ([Figma](https://www.figma.com/file/H3uTXxT88ofhdebUFgzEo9/PFT-%E2%80%93-Hand-off-Guide?node-id=3698%3A70349))
~Special section ([Figma](https://www.figma.com/file/H3uTXxT88ofhdebUFgzEo9/PFT-%E2%80%93-Hand-off-Guide?node-id=3698%3A69119))~
~Newsletter ([Figma](https://www.figma.com/file/H3uTXxT88ofhdebUFgzEo9/PFT-%E2%80%93-Hand-off-Guide?node-id=3698%3A62440))~

_ETA 2/14/22: Let's push off special section and newsletter designs for redesign phase 2_

### PR DESCRIPTION
**Pull request by SaptakS**
Fixes #1198 

This adds the blog details template, without the "In the Database" section. Also, fixed few JS fallbacks for the data table viewing and updated heading blocks.

### CODE REVIEW

File: blog/templates/blog/_details_table.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+		{% if blog.author %}
+		<div class="details-table__row">
+			<dt class="details-table__label details-table__label--bold">
+				Written by
+			</dt>
+			<dd class="details-table__value">
+				<a href="{% pageurl blog.author %}" class="text-link">
+					{{ blog.author.title }}
+				</a>
```
Comment (chigby) [inline blog/templates/blog/_details_table.html]: Sorry that this is kind of a nitpick, but putting this value `{{ blog.author.title }}` on a line by itself is causing the link to render with extra whitespace in it and the underline extends past the end of the text (for me, anyway). I have run into this problem a fair bit myself and the simplest solution is to condense these 3 lines into a single line. If there's another way, I'd actually like to hear about it, as other than `spaceless` (which doesn't really do what we want here) Django templates don't appear to me to have the kind of fine-grained whitespace controls that allow us to preserve readable formatting as well as getting the output we want.

File: blog/templates/blog/blog_page.html [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
-    {% endfor %}
+{% block main %}
+	<h1 class="article-title">{{ page.title }}</h1>
+
+	<article class="article-content">
+		<!-- Details table-->
+		{% include "blog/_details_table.html" with blog=page only %}
+
+
+		{% if page.introduction %}
```
Comment (chigby) [inline blog/templates/blog/blog_page.html:12]: Does the blog page model have an introduction field?

File: blog/templates/blog/blog_page.html [hunk 3]
```diff
@@ -1,9 +1,27 @@
...
+				<p class="heading">{{ page.introduction }}</p>
+		{% endif %}
+
+		<section class="article-content__inner">
+			{% include 'common/_video_or_image.html' with video=page.primary_video image=page.teaser_image caption=page.image_caption %}
```
Comment (chigby) [inline blog/templates/blog/blog_page.html:17]: I am not sure this `video_or_image` component matches the design exactly, the font size is (I think) one step too large and it does not include the image attribution.

**Review by chigby — CHANGES_REQUESTED** [review]
Requested a few changes as noted in the comments. I also opened a PR (#1254) that will, once merged and this branch rebased, will resolve the failing safety check.

Let me know if you have any questions.

**Comment by SaptakS** [comment]
@chigby addressed your comments

**Review by chigby — APPROVED** [review]

**Pull request merged**